### ISSUE TITLE
Driver is not thread safe

### ISSUE THREAD

**Issue by wwwdata** [univedo/univedo-go#4]
While running some integration tests with another project that uses this driver, I detected a DATA RACE Condition. I guess the problem lies in the remote_object implementation. I think especially this line of code here is problematic:  https://github.com/univedo/univedo-go/blob/master/remote_object.go#L164

```
WARNING: DATA RACE
Write by goroutine 67:
  runtime.closechan()
      /usr/local/Cellar/go/1.4.2/libexec/src/runtime/chan.go:257 +0x0
  github.com/univedo/univedo-go.func·005()
      /Users/bbieker/Go/src/git.evomecs.com/evomecs/toolmanager/Godeps/_workspace/src/github.com/univedo/univedo-go/sql.go:189 +0x64

Previous read by goroutine 66:
  runtime.chansend()
      /usr/local/Cellar/go/1.4.2/libexec/src/runtime/chan.go:83 +0x0
  github.com/univedo/univedo-go.func·006()
      /Users/bbieker/Go/src/git.evomecs.com/evomecs/toolmanager/Godeps/_workspace/src/github.com/univedo/univedo-go/sql.go:201 +0x1b3

Goroutine 67 (running) created at:
  github.com/univedo/univedo-go.(*BasicRemoteObject).receive()
      /Users/bbieker/Go/src/git.evomecs.com/evomecs/toolmanager/Godeps/_workspace/src/github.com/univedo/univedo-go/remote_object.go:164 +0x228e
  github.com/univedo/univedo-go.(*result).receive()
      <autogenerated>:21 +0x8c
  github.com/univedo/univedo-go.(*Connection).handleWebsocket()
      /Users/bbieker/Go/src/git.evomecs.com/evomecs/toolmanager/Godeps/_workspace/src/github.com/univedo/univedo-go/connection.go:133 +0x872
  github.com/univedo/univedo-go.func·001()
      /Users/bbieker/Go/src/git.evomecs.com/evomecs/toolmanager/Godeps/_workspace/src/github.com/univedo/univedo-go/connection.go:61 +0x4a

Goroutine 66 (finished) created at:
  github.com/univedo/univedo-go.(*BasicRemoteObject).receive()
      /Users/bbieker/Go/src/git.evomecs.com/evomecs/toolmanager/Godeps/_workspace/src/github.com/univedo/univedo-go/remote_object.go:164 +0x228e
  github.com/univedo/univedo-go.(*result).receive()
      <autogenerated>:21 +0x8c
  github.com/univedo/univedo-go.(*Connection).handleWebsocket()
      /Users/bbieker/Go/src/git.evomecs.com/evomecs/toolmanager/Godeps/_workspace/src/github.com/univedo/univedo-go/connection.go:133 +0x872
  github.com/univedo/univedo-go.func·001()
      /Users/bbieker/Go/src/git.evomecs.com/evomecs/toolmanager/Godeps/_workspace/src/github.com/univedo/univedo-go/connection.go:61 +0x4a
```

**Comment by lucas-clemente** [comment]
https://github.com/univedo/univedo-go/blob/master/sql.go#L189 and https://github.com/univedo/univedo-go/blob/master/sql.go#L201 can race – that's indeed a bug.

The rationale here is that if a registered notification handler causes another call to the server, it would block unless it runs in an extra thread. This would happen e.g. with listeners. It's probably fine to put that in the docs and run it without a separate goroutine by default.

TLDR: Could you try removing the `go` in https://github.com/univedo/univedo-go/blob/master/remote_object.go#L164 and check whether the rest still works?

**Comment by wwwdata** [comment]
hm, removing the `go` from that line let's the test suite hang...

It's completely fine to let the database driver block the current goroutine. I would even assume that this is always the case.
